extensions/sql-database-projects/src/models/project.ts

```diff
@@ -1442,6 +1477,81 @@ export class Project implements ISqlProject {
 		}
 	}
 
+	/**
+	 * Adds a value to project property, separated by semicolon if
+	 * one or more values are already present in project property
+	 * @param propertyName Name of property
+	 * @param valueToAdd Value to be added to project property, values containing semicolon are not supported
+	 * @param caseSensitive Determines if value to add to collection should be compared against existing values case sensitively
+	 */
+	private async addValueToCollectionProjectProperty(propertyName: string, valueToAdd: string, caseSensitive?: boolean): Promise<void> {
+		if (valueToAdd.includes(';')) {
+			return;
+		}
+
+		let propertyValueToSet = this.evaluateProjectPropertyValue(propertyName);
+		if (propertyValueToSet === undefined) {
+			propertyValueToSet = valueToAdd;
+		} else {
+			const propertyValue = caseSensitive ? propertyValueToSet : propertyValueToSet.toLocaleLowerCase();
+			const originalValueToAdd = valueToAdd;
+			valueToAdd = caseSensitive ? valueToAdd : valueToAdd.toLocaleLowerCase();
+			if ((propertyValue.length > valueToAdd.length
+				&& (propertyValue.startsWith(`${valueToAdd};`)
+					|| propertyValue.endsWith(`;${valueToAdd}`)
+					|| propertyValue.indexOf(`;${valueToAdd};`) >= 0))
+				|| (propertyValue.length === valueToAdd.length
+					&& propertyValue === valueToAdd)) {
+				return;
+			}
+
+			propertyValueToSet = `${propertyValueToSet};${originalValueToAdd}`;
+		}
+
+		await this.setProjectPropertyValue(propertyName, propertyValueToSet);
+	}
+
+	/**
+	 * Removes value from project property where values are semicolon separated
+	 * @param propertyName Name of property
+	 * @param valueToRemove Value to remove from the project property
+	 * @param caseSensitive Determines if value to remove from collection should be compared case sensitively
+	 */
+	protected async removeValueFromCollectionProjectProperty(propertyName: string, valueToRemove: string, caseSensitive?: boolean): Promise<void> {
+		let propertyValueToSet = this.evaluateProjectPropertyValue(propertyName);
+		if (propertyValueToSet === undefined
+			|| propertyValueToSet.length < valueToRemove.length
+			|| valueToRemove.includes(';')) {
```

Comment:
For this case, throw an error saying that the passed value isn't valid, and let the caller handle that.  Don't think we should sneak that in with the optimization no-ops.